src/ty.md

```diff
@@ -107,13 +107,14 @@ or `fn(i32) -> i32` (with type aliases fully expanded).
 
 ## `ty::Ty` implementation
 
-[`rustc::ty::Ty`][ty_ty] is actually a type alias to [`&TyS`][tys] (more about that later). `TyS`
-(Type Structure) is where the main functionality is located. You can ignore `TyS` struct in general;
+[`rustc::ty::Ty`][ty_ty] is actually a type alias to [`&TyS`][tys]. This type `TyS` (for "Type
+Structure") is where the main functionality is located. You can ignore `TyS` struct in general;
 you will basically never access it explicitly. We always pass it by reference using the `Ty` alias.
 The only exception is to define inherent methods on types. In particular, `TyS` has a [`kind`][kind]
 field of type [`TyKind`][tykind], which represents the key type information. `TyKind` is a big enum
-which represents different kinds of types (e.g. primitives, references, abstract data types,
-generics, lifetimes, etc). `TyS` also has 2 more fields, `flags` and `outer_exclusive_binder`. They
+with variants to represent many different sorts of types
```

Comment:
```suggestion
with variants to represent many different of Rust types
```

Reply:
I actually think "sorts of" is better in this case

Reply:
isn't "sort" a synonym of "kind" and "type"

Reply:
Yes, that's point. "different kinds of types" == `TyKind`

Reply:
Except that I think we avoided using the word "kind" unnecessarily here because of its ambiguity with the type-theoretic term "kind"...

Reply:
I find it redundant, but not too attached